aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -57,6 +57,30 @@ struct nansum_functor {
   }
 };
 
+const char nansum_name[] = "nansum";
+template <>
+struct nansum_functor<c10::complex<at::Half>> {
+#if AT_USE_JITERATOR()
+  void operator()(TensorIterator& iter) {
+    using scalar_t = c10::complex<at::Half>;
+    std::string func = jiterator_stringify(
+        arg_t combine(arg_t a, scalar_t b) {
+          return a + ((std::isnan(b.real()) || std::isnan(b.imag())) ? arg_t{0.} : arg_t{b});
```

Comment:
std::is_nan got support for c10:complex numbers in this PR: https://github.com/pytorch/pytorch/pull/93153

Reply:
Great point. I think the Jiterator complex support is used from the llvm_complex.cpp file. I almost forgot this and used a naive approach. Thanks, Aaron! I've added `isnan` op there and updated it here.

Reply:
Since you seem to be an expert on this, any idea why the CUDA op generated by the CUDA file the OG Logcumsumexp PR ended up taking half an hour to compile? https://github.com/pytorch/pytorch/pull/90847 , getting Logcumsumexp to have CUDA support too would be nice as well since most of the work is already done.